File: jiheekim/041_예산.cpp
#include <iostream>
#include <stdio.h>
#include <string>
#include <vector>
#include<algorithm>

using namespace std;

int solution(vector<int> d, int budget) {
int sum=0;
    int temp=budget;
 sort(d.begin(), d.end());

    for (int i=0; i<d.size();i++){
       budget=budget-d[i];
       sum=sum+d[i];
  
        if(sum>temp)
                return i;
    }

}

File: lesunny/041_예산.java
import java.util.Arrays;

class Solution {
    public int solution(int[] d, int budget) {
        int answer = 0;
        
        Arrays.sort(d);
        
        int i=0;
        while( i<d.length && budget >= d[i]){
            answer++;
            budget -= d[i++];
        }
        
        return answer;
    }
}
